docs/madr/decisions/049-meshexternalservice-api.md

```diff
@@ -0,0 +1,474 @@
+# MeshExternalService API
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/6330
+
+## Context and Problem Statement
+
+The current implementation of `ExternalService` has significant limitations and warrants a complete redesign.
+
+* Lack of support for * subdomains
+* Lack of support for thick clients (think Kafka or C* where discovery is done as part of the client).
+* Need for pluggability (for example lambda support or equivalent)
+* Working well with gateways (what if your ExternalService relies on SNI/host header to route through a gateway)
+* Applying policies
+
+
+## Considered Options
+
+* Creating MeshExternalService + HostnameGenerator
+* Creating MeshExternalService
+* Creating MeshExternalService and using HostnameGenerator for InternalVIP
+
+## Decision Outcome
+
+Chosen option: "Creating MeshExternalService".
+
+### Positive Consequences
+
+* Clearer structure.
+* Enhanced functionalities.
+* Better policy matching capabilities.
+* No need for a HostnameGenerator
+
+### Negative Consequences
+
+* Whole new resource with a lot of new code.
+* Temporarily increased complexity of a product until the migration is done.
```

Comment:
have we decided on a migration path? how would it look like? if we have an ExternalService and MeshExternalService targeting the same destination MES will take over?

Reply:
Migration added at the end